### PR TITLE
Generate default error message

### PR DESCRIPTION
**Pull request by PhilipIngham**
## Overview

I have added a default error message for the [Assertions.fail()](https://github.com/junit-team/junit5/blob/1c294651ab3965a8bfe8909f6efdc56c5a47c4f0/junit-jupiter-api/src/main/java/org/junit/jupiter/api/Assertions.java#L106-L118) function as discussed in the issue, and made some grammatical changes in the surrounding areas. As said in the issue, it's a very minor problem, so nothing is really lost if this doesn't get implemented, but the option's there.
---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit5/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [ ] There are no TODOs left in the code
- [ ] Method [preconditions](https://junit.org/junit5/docs/snapshot/api/org.junit.platform.commons/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [ ] [Coding conventions](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [ ] Change is covered by [automated tests](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [ ] Public API has [Javadoc](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [ ] Change is documented in the [User Guide](https://junit.org/junit5/docs/snapshot/user-guide/) and [Release Notes](https://junit.org/junit5/docs/snapshot/user-guide/#release-notes)

### CODE REVIEW

File: junit-platform-engine/src/main/java/org/junit/platform/engine/reporting/ReportEntry.java [hunk 1]
```diff
@@ -83,7 +83,7 @@ private void add(String key, String value) {
 	 *
 	 * @return a copy of the map of key-value pairs; never {@code null}
 	 */
-	public final Map<String, String> getKeyValuePairs() {
+	public Map<String, String> getKeyValuePairs() {
```
Comment (davidgouldbjss) [inline junit-platform-engine/src/main/java/org/junit/platform/engine/reporting/ReportEntry.java]: Apologies if I'm missing something: does this address a particular issue, as in does something need to override these methods as I can't see how this relates to the description or overview in the PR?

Thanks

Comment (PhilipIngham) [reply junit-platform-engine/src/main/java/org/junit/platform/engine/reporting/ReportEntry.java]: Can't remember why I did that tbh, I think it was related to the class being final, so the IDE was complaining that the functions within are final. Stupid change, shall remove 🤦

Comment (PhilipIngham) [reply junit-platform-engine/src/main/java/org/junit/platform/engine/reporting/ReportEntry.java]: Re-added it back in, along with the getTimestamp() function

**Review by davidgouldbjss — APPROVED** [review]

**Review by davidgouldbjss — APPROVED** [review]
